network-api/networkapi/wagtailpages/pagemodels/research_hub/authors_index.py

```diff
@@ -81,3 +81,12 @@ def get_author_detail_context(self, profile_id: int):
             'author_profile': author_profile,
             'latest_research': latest_research,
         }
+
+    def get_breadcrumbs(self):
```

Comment:
Hi @tbrlpld! Per our conversation on slack, I have updated the pages to have their own "get_breadcrumbs" method, which gets the pages ancestors, and then trims backwards by reversing the list before trimming. 

One thing I am still working on, is getting the author detail page to render the "authors" breadcrumb while not having the author index page display it.

Also, I was testing this method that works for all pages without having to trim the query, in case you might prefer using this instead. Though I am not sure if calling get_ancestors twice is best practice:
```
    def get_breadcrumbs(self):
        research_landing_page = self.get_ancestors().type(landing_page.ResearchLandingPage).first()
        page_parents = self.get_ancestors().descendant_of(research_landing_page, True)
        breadcrumb_list = [{"title": parent_page.title, "url": parent_page.url} for parent_page in page_parents]

        return breadcrumb_list
```

Reply:
You don't need to reverse and then trim. In Python you can trim from the end by using negative indexes. 

```python
>>> ["a", "b", "c"][-2:]
['b', 'c']

```

Reply:
Since we actually never want the "author route" (aka. author detail page) in the breadcrumb, but only the author index page if we are on a "author route", I think you can solve this by adding a `inclusive` parameter to the `get_breadcrumbs` method which you pass to the `get_ancestors` method. 

This parameter determines if the page it self is included in the ancestors. When on a "author route" you set this to `True` which will include the author index page in the breadcrumbs.

Reply:
**Kudos** I really like this approach of filtering for descendants of the landing page rather than trimming the list  :+1:

Reply:
**Suggestion** If you rebase, you will see that there is now a ResearchHubBasePage. You can add the `get_breadcrumbs` method to that so we don't have to keep these 3 page types in sync if anything changes. If you need to invoke the functions differently, you can do that in the `get_context` method of each page.

Reply:
**Nitpick** Since we are calling the `get_ancestors` method, lets pass it to a variable called `ancestors` or `page_ancestors` since parents are usually only one level up (but we are dealing with multiple levels).

Reply:
**Suggestion** I think it would be great if there were some tests around the breadcrumbs generation. That will help us detect changes of breakage in the future.